### PR TITLE
Make output file an explicit argument.

### PR DESCRIPTION
**Pull request by lurch**
(because relying on output-redirection meant that on failure, it still "sucessfully" created a zero-byte output file, which messed up later Make steps)
Also update requirements.txt

### CODE REVIEW

**Comment by lurch** [comment]
I've got corresponding changes to the other repos that rely on _this_ repo as a submodule, which I'll PR after this has been merged.

File: cli.py [hunk 1]
```diff
@@ -8,28 +8,28 @@
 
 
 def main():
-    """Convert the given Doxygen index.xml to AsciiDoc and print the result."""
+    """Convert the given Doxygen index.xml to AsciiDoc and output the result."""
     parser = argparse.ArgumentParser()
     parser.add_argument("-f", "--file", help="The path of the file to convert", default=None)
+    parser.add_argument("-o", "--output", help="The path of the output file", default=None)
```
Comment (mudge) [inline cli.py:14]: For backward compatibility (and to behave like curl), would it be worth defaulting to STDOUT if no output file is specified?

Comment (lurch) [reply cli.py:14]: Ah, good call. Done :slightly_smiling_face:

File: cli.py [hunk 2]
```diff
@@ -8,28 +8,31 @@
...
-                )
-                
+                result = DoxygenindexNode(
+                    BeautifulSoup(xml, "xml").doxygenindex, xmldir=xmldir
+                ).to_asciidoc()
+        if output_filename is not None:
+            with open(output_filename, "w", encoding="utf-8") as output:
+                output.write(result)
+        else:
+            print(result)
```
Comment (mudge) [inline cli.py:35]: What happens if someone passes `-o` or `--output` with no argument? Does it come through as an empty string or as `None`?

If the former, would it be safer to use the inherit "truthiness" of a non-empty string like so:

```suggestion
        if output_filename:
            with open(output_filename, "w", encoding="utf-8") as output:
                output.write(result)
        else:
            print(result)
```

Comment (lurch) [reply cli.py:35]: > What happens if someone passes `-o` or `--output` with no argument?

Good question. I just did a quick test, and it automatically complains with `error: argument -o/--output: expected one argument` and prints the usage-string :smiley:

**Review by mudge — APPROVED** [review]

**Review by nathan-contino — APPROVED** [review]
LGTM!

**Pull request merged**